util/udma_barrier.h

```diff
@@ -201,7 +201,7 @@
 #elif defined(__sparc_v9__)
 #define mmio_flush_writes() asm volatile("membar #StoreStore" ::: "memory")
 #elif defined(__aarch64__)
-#define mmio_flush_writes() asm volatile("dsb st" ::: "memory");
+#define mmio_flush_writes() asm volatile("dmb oshst" ::: "memory");
```

Comment:
We should use DGH here. But, suggest to make this in another commit as it would need a different commit log.

Reply:
I think you should take the mmio_flush_writes change out of this commit and revisit it in a followup. The justification you gave for this change doesn't apply here. DGH alone may also not be correct.

This code needs to ensure that a write combining is entirely contained within a spinlock region. No write combine can be merged across different spinlock critical sections. AFAIK this requirement has nothing to do with other-multi-copy-atomicity.

Reply:
Thank you for the review comments! I have updated the PR to remove the mmio_flush_writes change, will revisit in a follow up.